File: backend/app/schemas/baseline.py
from datetime import datetime
from typing import Any, List, Optional
from uuid import UUID
from pydantic import BaseModel, Field


class MeasurementUploadDataOut(BaseModel):
    id: UUID
    upload_id: UUID
    sensor_id: UUID
    original_filename: str
    file_format: str
    channel_count: int
    sample_count: int
    created_at: datetime

    model_config = {"from_attributes": True}


class BaselineCreateFromUpload(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    description: Optional[str] = None
    labels: List[str] = Field(default_factory=list)
    set_as_primary: bool = False
    captured_at: Optional[datetime] = None


class BaselineOut(BaseModel):
    id: UUID
    sensor_id: UUID
    source_upload_id: Optional[UUID]
    name: str
    description: Optional[str]
    labels: List[str]
    original_filename: str
    file_format: str
    channel_count: int
    sample_count: int
    sampling_rate_hz: float
    is_primary: bool
    captured_at: Optional[datetime]
    created_at: datetime
    plot_count: int = 0
    plots_status: str = "unknown"

    model_config = {"from_attributes": True}


class BaselineSetPrimary(BaseModel):
    is_primary: bool = True


class BaselineListOut(BaseModel):
    sensor_id: UUID
    total: int
    primary_baseline_id: Optional[UUID]
    items: List[BaselineOut]

File: frontend/src/types/baseline.ts
export interface Baseline {
  id: string;
  sensor_id: string;
  source_upload_id: string | null;
  name: string;
  description: string | null;
  labels: string[];
  original_filename: string;
  file_format: string;
  channel_count: number;
  sample_count: number;
  sampling_rate_hz: number;
  is_primary: boolean;
  captured_at: string | null;
  created_at: string;
  plot_count: number;
  plots_status: string;
}

export interface BaselineListResponse {
  sensor_id: string;
  total: number;
  primary_baseline_id: string | null;
  items: Baseline[];
}

export interface BaselineCreateFromUpload {
  name: string;
  description?: string | null;
  labels?: string[];
  set_as_primary?: boolean;
  captured_at?: string | null;
}
